Sources/TSCLibc/libc.swift

```diff
@@ -20,44 +20,91 @@
 @_exported import TSCclibc
 
 #if os(Windows)
+private extension String {
+    func withCStringW<Result>(_ body: (UnsafePointer<wchar_t>, Int) throws -> Result) rethrows -> Result {
+        return try withCString(encodedAs: UTF16.self) {
+            let capacity: Int = wcslen($0) + 1
+            return try $0.withMemoryRebound(to: wchar_t.self, capacity: capacity) {
+                try body($0, capacity)
+            }
+        }
+    }
+}
+
+private extension UnsafeMutablePointer where Pointee == CChar {
+    func assign(from source: UnsafePointer<wchar_t>) {
+        String(decodingCString: source, as: UTF16.self).utf8CString.withUnsafeBytes {
+            assign(from: $0.bindMemory(to: CChar.self).baseAddress!,
+                count: $0.count)
+        }
+    }
+}
+
 // char *realpath(const char *path, char *resolved_path);
 public func realpath(
     _ path: String,
     _ resolvedPath: UnsafeMutablePointer<CChar>?
 ) -> UnsafeMutablePointer<CChar>? {
-  fatalError("realpath is unimplemented")
+    let result: UnsafeMutablePointer<CChar>
+    if let resolvedPath = resolvedPath {
+        result = resolvedPath
+    } else {
+        result = UnsafeMutablePointer<CChar>.allocate(capacity: Int(MAX_PATH))
+    }
+    return String(cString: result).withCStringW { resultW, capacity in
+        return path.withCStringW { pathW, _ in
+            guard _wfullpath(UnsafeMutablePointer(mutating: resultW), pathW, capacity) != nil else {
```

Comment:
I don't think that this is really the equivalent to `realpath` btw, this will expand the relative paths but I don't think it will resolve symlinks along the way.